pypeit/coadd2d.py

```diff
@@ -561,6 +561,11 @@ def reduce(self, pseudo_dict, show=None, show_peaks=None, basename=None):
             manual_string = ','.join(np.atleast_1d(self.par['coadd2d']['manual']))
             if len(manual_string.strip()) > 0:
                 manual_obj = ManualExtractionObj.by_fitstbl_input('None', manual_string)
+                uniq_dets = np.unique(manual_obj.det)
+                if uniq_dets.size > 1:
+                    msgs.error('2D co-adding does not support extractions from multiple detectors. '
+                               'Perform the co-adding for each detector separately.')
+
```

Comment:
@jhennawi I wrote this piece of code to allow users to put manual extractions in different detectors when they are running `pypeit_coadd_2dspec` (which  loops over the detectors).  If you add this error, this functionality is basically removed. Here, the code is selecting only the manual extraction parameters relevant to the detector that is currently reduced. I know you are going to say that this needs to be done upstream and I agree. If you want I can take care of. Just let me know.

Reply:
I'm not actually sure why I put this in there. But the top level question is, how do we want to approach 2d co-adding for multiple detectors? It seems to me we want separate coadd2d files, but I have not thought it through. Let's discuss that point.

Reply:
Let's discuss. I've commented it out for now in a future PR when I merged DEIMOS mosaic in.